lib/models/device.ts

```diff
@@ -3234,11 +3188,17 @@ const getDeviceModel = function (
 			// this will throw an error if the action isn't available
 			exports._checkOsUpdateTarget(device, targetOsVersion);
 
-			const { versions: allVersions } = await osModel().getSupportedVersions(
-				device.is_of__device_type[0].slug,
-			);
-
-			if (!allVersions.some((v) => bSemver.compare(v, targetOsVersion) === 0)) {
+			const osVersions = (
+				await hostappModel().getAllOsVersions([
```

Comment:
Reminder that we also need the new hostAppModel to be used/offer something like `os.getSupportedOsUpdateVersions` (the CLI uses this).